modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SpringBootServerCodegen.java

```diff
@@ -119,9 +131,6 @@ public void addOperationToGroup(String tag, String resourcePath, Operation opera
         if (basePath == "") {
             basePath = "default";
         } else {
-            if (co.path.startsWith("/" + basePath)) {
-                co.path = co.path.substring(("/" + basePath).length());
-            }

```

Comment:
It's because if you put a RequestMapping annotation on the interface class then Spring-MVC will make a controller of any implementation bean (even if it's not a @Controller) and this is undesirable in use cases such as spring-cloud-feign. This commit removes the RequesMapping on the interface and puts the path segment into the methods. In my opinion this is also better for controllers since you can use RequestMapping on the controller to configure the basePath (eg. "v2" in petstore).